src/topology.md

```diff
@@ -27,7 +41,12 @@ intermediate mixnodes should be uniformly randomly chosen, subject to the follow
 - No mixnode should appear in a route more than once, unless this is unavoidable. For example, when
   a non-mixnode is generating a loop route, if there is only a single connected gateway mixnode,
   the gateway must necessarily appear twice.
+  // TODO in this case I would find it better to return an error on route generation and put back
+  // the mixnet in a synching state.
+  // and state that this rule cannot be check by relaying node (only when creating route).
 - No node should ever appear twice consecutively in a route.
+  // TODO maybe state that this rule can be enforced by relaying node by simply dropping the packet
```

Comment:
These rules are really just to avoid poor anonymity of particular packets due to poor route choice. I don't think there's any need for mixnodes to attempt to enforce them; if a node wants to compromise the anonymity of the packets it sends then there are many ways for it to do so.

FWIW this rule is actually almost redundant given the "no mixnode should appear in a route more than once" rule.

Reply:
> FWIW this rule is actually almost redundant given the "no mixnode should appear in a route more than once" rule.

it is (but the "unless it is unavoidable" makes things awkward i think)